pkg/resourcetotelemetry/resource_to_telemetry.go

```diff
@@ -54,28 +64,59 @@ func WrapMetricsExporter(set Settings, exporter component.MetricsExporter) compo
 	if !set.Enabled {
 		return exporter
 	}
-	return &wrapperMetricsExporter{MetricsExporter: exporter}
+	return &wrapperMetricsExporter{MetricsExporter: exporter, MatchAttributes: set.Exclude}
 }
 
-func convertToMetricsAttributes(md pmetric.Metrics) pmetric.Metrics {
+func convertToMetricsAttributes(md pmetric.Metrics, filter MatchAttributes) pmetric.Metrics {
 	cloneMd := pmetric.NewMetrics()
 	md.CopyTo(cloneMd)
 	rms := cloneMd.ResourceMetrics()
 	for i := 0; i < rms.Len(); i++ {
 		resource := rms.At(i).Resource()
 
+		labelMap := extractLabelsFromResource(&resource, filter)
+
 		ilms := rms.At(i).ScopeMetrics()
 		for j := 0; j < ilms.Len(); j++ {
 			ilm := ilms.At(j)
 			metricSlice := ilm.Metrics()
 			for k := 0; k < metricSlice.Len(); k++ {
-				addAttributesToMetric(metricSlice.At(k), resource.Attributes())
+				addAttributesToMetric(metricSlice.At(k), labelMap)
 			}
 		}
 	}
 	return cloneMd
 }
 
+// extractAttributesFromResource extracts the attributes from a given resource and
+// returns them as a StringMap.
+func extractLabelsFromResource(resource *pcommon.Resource, filter MatchAttributes) pcommon.Map {
+	labelMap := pcommon.NewMap()
+	//var err error
+
+	if len(filter.Exclude) == 0 {
+		return resource.Attributes()
+	}
+
+	attrMatcher := filtermatcher.AttributesMatcher{}
+	attrMatcher, _ = filtermatcher.NewAttributesMatcher(filter.Filter, filter.Exclude)
```

Comment:
We should build this on `WrapMetricsExporter` so that we can reuse it across calls. The error should also be handled; that's going to make things a bit more difficult since it is a breaking change. You should follow the guidelines [here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#example-2---changing-the-return-values-of-a-function)

Reply:
fixed